content/chapters/compute/lecture/quiz/exec-without-fork.md

```diff
@@ -0,0 +1,22 @@
+## Question Text
+
+We type the following command:
+```bash
+student@os:~$ ls
+```
+Why does the shell **NOT** simply call `exec("/bin/ls")`?
+
+![Bash exec](../media/exec.svg)
+
+## Question Answers
+
+- Because `/bin/ls` may be buggy or vulnerable and compromise the sell
++ Because when `ls` ends, the shell is closed
+- Because the `ls` and `bash` processes must not share the same address space
+- Because `bash` needs to pass data to the `ls` process
+
+## Feedback
+
+If the shell simply `exec`s `"/bin/ls"`, the `bash` process is entirely replaced by the `ls` process.
```

Comment:
I think there is never mentioned what `exec` does (replacing the process image).

Reply:
Shouldn't this be covered in the Data chapter?

Reply:
The Data chapter explains `exec`, while the Compute chapter explains `fork`. Briefly reviewing `exec` after understanding `fork` and discussing how they are used together in practice feels natural for me.

Reply:
Yeah, you're right. I see your point now.

Reply:
I've added a few lines regarding `exec`.